usaspending_api/awards/v2/data_layer/orm.py

```diff
@@ -256,31 +253,43 @@ def fetch_award_details(filter_q: dict, mapper_fields: OrderedDict) -> dict:
     return Award.objects.filter(**filter_q).values(*vals).annotate(**ann).first()
 
 
-def fetch_parent_award_from_piid_agency(piid, fpds_agency):
-    if piid and fpds_agency:
-        parent_unique_key = "CONT_IDV_{}_{}".format(piid, fpds_agency)
-        parent = (
-            ParentAward.objects.filter(generated_unique_award_id=parent_unique_key)
-            .values("generated_unique_award_id")
-            .first()
-        )
-        if parent:
-            return parent["generated_unique_award_id"]
-    return None
+def fetch_contract_parent_award_details(parent_piid: str, parent_fpds_agency: str) -> Optional[OrderedDict]:
+    parent_guai = "CONT_IDV_{}_{}".format(parent_piid or "NONE", parent_fpds_agency or "NONE")
```

Comment:
I would advise against us getting used to using `guai` to mean `generated unique award id`. It's not exactly obvious.

Reply:
`unique_id` `gen_unique_id` `generated_id`

`GUAI` is more clearly an acronym.